internal/terraform/datadir/module_types.go

```diff
@@ -0,0 +1,36 @@
+package datadir
+
+import (
+	"github.com/hashicorp/go-getter"
+	tfregistry "github.com/hashicorp/terraform-registry-address"
+)
+
+type ModuleType string
+
+const (
+	NOTFOUND   ModuleType = "notfound"
+	TFREGISTRY ModuleType = "tfregistry"
+	LOCAL      ModuleType = "local"
+	GITHUB     ModuleType = "github"
+	GITLAB     ModuleType = "gitlab"
+)
+
+func (r *ModuleRecord) GetModuleType() ModuleType {
+	if _, err := tfregistry.ParseRawProviderSourceString(r.SourceAddr); err == nil {
+		return TFREGISTRY
+	}
+	if _, ok, _ := new(getter.GitHubDetector).Detect(r.SourceAddr, ""); ok {
+		return GITHUB
+	}
+	if _, ok, _ := new(getter.GitLabDetector).Detect(r.SourceAddr, ""); ok {
+		return GITLAB
+	}
+	if _, ok, _ := new(getter.GitDetector).Detect(r.SourceAddr, ""); ok {
+		return GITHUB
```

Comment:
```suggestion
		return GIT
```
Typo? 👀

Reply:
This was a placeholder until we could decide on icon for git repo based modules. We have landed on `git-branch`, so we can now correctly report the type